Bundle/WidgetBundle/Model/WidgetManager.php

```diff
@@ -174,27 +134,19 @@ public function createWidget($mode, $type, $slotId, View $view, $entity, $positi
             //update fields of the widget
             $widget->setBusinessEntityId($entity);
 
-            $widget->positionReference = $positionReference;
-            $widget->slotId = $slotId;
             //persist the widget
             $this->entityManager->persist($widget);
             $this->entityManager->flush();
 
-            //create the new widget map
-            $widgetMapEntry = new WidgetMap();
-            $widgetMapEntry->setAction(WidgetMap::ACTION_CREATE);
-            $widgetMapEntry->setWidgetId($widget->getId());
-
-            $widgetMap = $this->widgetMapBuilder->build($view, false);
-
-            $widgetMapEntry = $this->widgetMapPositionBuilder->generateWidgetPosition($this->entityManager, $widgetMapEntry, $widget, $widgetMap, $positionReference);
-            $this->widgetMapHelper->insertWidgetMapInSlot($slotId, $widgetMapEntry, $view);
+            $this->widgetMapManager->insert($widget, $view, $slotId, $position, $widgetReference);
 
             $this->entityManager->persist($view);
             $this->entityManager->flush();
 
             $widget->setCurrentView($view);
```

Comment:
please have a look on the **currentView** property, I'd rather to always use the currentView Helper